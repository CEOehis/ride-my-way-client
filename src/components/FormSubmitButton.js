import React from 'react';
import PropTypes from 'prop-types';

const FormSubmitButton = ({ submitting }) => (submitting
  ? <input className="btn btn-lg btn-submit btn-orange submitting" type="submit" value="submitting" disabled />
  : <input className="btn btn-lg btn-submit btn-orange" type="submit" value="submit" />);


FormSubmitButton.propTypes = {
  submitting: PropTypes.bool.isRequired,
};

export default FormSubmitButton;
