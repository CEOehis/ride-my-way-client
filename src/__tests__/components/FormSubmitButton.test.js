import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FormSubmitButton from '../../components/FormSubmitButton';

describe('Form submit button component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<FormSubmitButton submitting />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
