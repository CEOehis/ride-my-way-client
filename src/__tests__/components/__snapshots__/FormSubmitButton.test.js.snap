// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form submit button component should render correctly 1`] = `
<input
  className="btn btn-lg btn-submit btn-orange submitting"
  disabled={true}
  type="submit"
  value="submitting"
/>
`;
